### PR TITLE
feat(connector): [Gocardless] add boilerplate code

### PR DESCRIPTION
**Pull request by ArjunKarthik**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Gocardless new connector integration

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/test_utils/src/connector_auth.rs [hunk 1]
```diff
@@ -76,7 +77,7 @@ impl ConnectorAuthentication {
     /// Will panic if `CONNECTOR_AUTH_FILE_PATH` env is not set
     #[allow(clippy::expect_used)]
     pub fn new() -> Self {
-        // Do `export CONNECTOR_AUTH_FILE_PATH="/hyperswitch/crates/router/tests/connectors/sample_auth.toml"`
+        // Do `export CONNECTOR_AUTH_FILE_PATH="/hyperswitch/crates/router/tests/connectors/sample_sample_auth.toml"`
```
Comment (ArjunKarthik) [inline crates/test_utils/src/connector_auth.rs:80]: This was added by the script, Will revert in next PR<

File: crates/router/src/core/payments/flows.rs [hunk 2]
```diff
@@ -142,6 +142,7 @@ impl<const T: u8>
 }
 
 default_imp_for_complete_authorize!(
+    connector::Gocardless,
```
Comment (prasunna09) [inline crates/router/src/core/payments/flows.rs:145]: Please maintain the order here as well as in the other default implementations.

**Review by SamraatBansal — APPROVED** [review]

File: crates/router/src/core/payments/flows.rs (hunk 2 above)
Comment (ArjunKarthik) [reply crates/router/src/core/payments/flows.rs:145]: @prasunna09 I resolved it already a separate PR -https://github.com/juspay/hyperswitch/pull/2180

File: crates/router/tests/connectors/gocardless.rs [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+
+#[derive(Clone, Copy)]
+struct GocardlessTest;
+impl ConnectorActions for GocardlessTest {}
+impl utils::Connector for GocardlessTest {
+    fn get_data(&self) -> types::api::ConnectorData {
+        use router::connector::Gocardless;
+        types::api::ConnectorData {
+            connector: Box::new(&Gocardless),
+            connector_name: types::Connector::Gocardless,
```
Comment (prasunna09) [inline crates/router/tests/connectors/gocardless.rs:18]: Please use `DummyConnector1`to fix the failed clippy check

**Review by prasunna09 — CHANGES_REQUESTED** [review]

**Review by prasunna09 — APPROVED** [review]

**Review by SamraatBansal — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**